src/App.js

```diff
@@ -65,7 +67,7 @@ class Machine extends Component {
 
     handleChange(event) {
         this.setState({source: event.target.value});
-        history.replaceState({source: event.target.value}, 'source', '?s=' + event.target.value);
+        history.replaceState({source: this.state.source, speed: this.state.speed}, 'Colorfuck', '?s=' + this.state.source + "&speed=" + this.state.speed );
```

Comment:
`setState()` is an asynchronous  operation, so accessing `this.state` immediately afterwards might still give the old value. We'll either have to use `event.target.value` or move the `history.replace state(…)` to a callback function passed to `setState` as a second argument